crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1517,6 +1517,14 @@ impl ObjectStore for AuthorityStore {
     fn get_object(&self, object_id: &ObjectID) -> Result<Option<Object>, SuiError> {
         self.perpetual_tables.as_ref().get_object(object_id)
     }
+
+    fn get_object_by_key(
```

Comment:
why not use an existing method?
https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority/authority_store.rs#L387

Reply:
I think that is what we are using and moved here. 
It's just we need to put that function in a trait as there is a lot of trait-shit going on that forces us to move things around...
let me know

Reply:
reviewed and you are right, good call. Removed my previous comment and fixed it!